runtime/lua/vim/secure.lua

```diff
@@ -1,5 +1,47 @@
 local M = {}
 
+--- @private
+--- Reads trust database from $XDG_STATE_HOME/nvim/trust.
+---
+--- @return (table) Contents of trust database, if it exists. Empty table otherwise.
+local function read_trust()
+  local trust = {}
+  local f = io.open(vim.fn.stdpath('state') .. '/trust', 'r')
+  if f then
+    local contents = f:read('*a')
+    if contents then
+      for line in vim.gsplit(contents, '\n') do
+        local hash, file = string.match(line, '^(%S+) (.+)$')
+        if hash and file then
+          trust[file] = hash
+        end
+      end
+    end
+    f:close()
+  end
+  return trust
+end
+
+--- @private
+--- Writes provided {trust} table to trust database at
+--- $XDG_STATE_HOME/nvim/trust.
+---
+--- @param trust (table) Trust table to write
+local function write_trust(trust)
+  vim.validate({ trust = { trust, 't' } })
+  local f, err = io.open(vim.fn.stdpath('state') .. '/trust', 'w')
+  if not f then
+    error(err)
+  end
```

Comment:
This can be simplified to

```suggestion
  local f = assert(io.open(vim.fn.stdpath('state') .. '/trust', 'w'))
```

Reply:
done in 13e2ad1aea4c569aa8cbc13e7e0016845bf2d765